docs/proposals/scheduler/scheduler.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+
+package scheduler;
+import "google/protobuf/empty.proto";
+
+enum Status {
+    RUNNING = 0;
+    STARTING = 1;
+    STOPPED = 2;
+    STOPPING = 3;
+    DESTROYING = 4;
+    TERMINATED = 5;
+    CRASHED = 6;
+    FAILED = 7;
+    SCHEDULING = 8;
+    SCHEDULED = 9;
+}
+
+enum Action {
+    START = 0;
+    STOP = 1;
+    DESTROY = 2;
+    KILL = 3;
+}
+
+enum Type {
+    CONTAINER = 0;
+}
+
+message Instance {
+    string id = 1;
+    string name = 2;
+    Type type = 3;
+    Status status = 4;
+    string uri = 5;
+    repeated string environnement = 6;
+    Resource resource = 7;
+    repeated string ports = 8;
+    string ip = 9;
+}
+
+message ResourceSummary {
+    int32 cpu = 1;
+    int32 memory = 2;
+    int32 disk = 3;
+}
+
+message Resource {
+    ResourceSummary max = 1;
+    ResourceSummary usage = 2;
+}
+
+message InstanceStatus {
+    string id = 1;
+    Status status = 2;
+    string description = 3;
+}
+
+message NodeStatus {
+    string id = 1;
+    Status status = 2;
+    string description = 3;
+    Resource resource = 4;
+    repeated Instance instances = 5;
+}
+
+message InstanceAction {
+    string id = 1;
+    Action action = 2;
+}
+
+message NodeRegisterRequest {
+    string certificate = 1;
+}
+
+message NodeRegisterResponse {
+    int32 code = 1;
+    string description = 2;
+    string subnet = 3;
+}
+
+service AgentService {
```

Comment:
That is your `NodeService`, which happens to be used by the agent on a node.

And it should only have a `NodeRegister` and `NodeUpdate`methods. Instances status updates should come from a stream returned when the scheduler sent an `InstanceCreate` request to the agent. See https://github.com/dev-sys-do/kudo/pull/4#discussion_r867690932

Reply:
In this case, the same behavior should be applied for the registration of a node (We return a stream that receives `NodeUpdate`) ?

Otherwise, is the `NodeUpdate` the same as `NodeStatusUpdate` ?

Reply:
> In this case, the same behavior should be applied for the registration of a node (We return a stream that receives `NodeUpdate`) ?

What would a `NodeStatus`stream provide that the combined agent status streams do not? The scheduler is already going to get all instances status updates through the per-instance streams that it gets from the agent.

At least for now I'd keep things simple and have node service only prrovide a Registration API (`Register`/`Unregister`).

Reply:
Ok we've changed.